ptxcompiler/patch.py

```diff
@@ -91,13 +91,28 @@ def get_cubin(self, cc=None):
         return cubin
 
 
+# Determine the driver and runtime versions for comparison. The logic here is a
+# little odd because the WSL2 preview driver (510.06 at the time of writing)
+# reports a CUDA version of 11.6 but it only accepts PTX up to version 7.4,
+# which is the maximum supported version for 11.4 - see the table in:
```

Comment:
Should we be directly checking the version of PTX it accepts instead (assuming that can be checked)?

Reply:
I think the way to check is to use `cuDriverGetVersion`, but it's (in my best estimation) not providing a correct result in the WSL2 driver.